src/compiler/scala/tools/nsc/typechecker/Implicits.scala

```diff
@@ -986,14 +986,21 @@ trait Implicits {
       /** Sorted list of eligible implicits.
        */
       val eligible = Shadower.using(isLocalToCallsite){ shadower =>
-        val matches = iss flatMap { is =>
-          val result = is filter (info => checkValid(info.sym) && survives(info, shadower))
+        var matches: mutable.ListBuffer[ImplicitInfo] = null
```

Comment:
This `var`/`null` trick isn't likely saving any allocations (we need to box it in an `ObjectRef`). Check if the optimizer/hotspot can indeed eliminate the boxing, or just make it a `val`.